bug: missing @babel/runtime dependency in yarn modern with pnp linker

**Describe the bug**

In yarn modern (version 2+) with the pnp linker, tests and build fails because helpers provided by @babel/runtime are missing. I believe @babel/runtime gets hoisted from another dependency (possibly aria-query according to `yarn why @babel/runtime`) in regular npm/yarn v1 so it seems to work. Yarn modern is more strict with dependency resolution, so I see failures after importing stream-chat-react components into a project and trying to run tests or build.

While I can also fix this by providing @babel/runtime myself, I don't think this is the "correct" solution. Rather, the solution should be to include @babel/runtime as a dependency in this project. From what I can tell, this is what is recommended when using @babel/plugin-transform-runtime according to the babel docs (unless I am misinterpreting what is written there, but I see a lot of other libraries that supply their own versions of @babel/runtime). See the the "technical details" section of babel-plugin-transform-runtime (https://babeljs.io/docs/babel-plugin-transform-runtime#technical-details):

> Make sure you include @babel/runtime as a dependency.

I am happy to submit a PR to fix this issue. And would love to hear your thoughts!

**To Reproduce**

I've created a bug repro repo here: https://github.com/imccausl/stream-chat-react-yarn-modern-bug-repro-repo

Steps to reproduce the behaviour:
1. Clone the project (https://github.com/imccausl/stream-chat-react-yarn-modern-bug-repro-repo)
2. install the project `nvm use && yarn install` (uses yarn v4 with the pnp linker)
3. `yarn build` to "build" the project with babel and webpack. You'll get an error:  `Can't resolve '@babel/runtime/helpers/defineProperty`
4. `yarn test` to run tests (transpiled with babel). You'll get an error: `Cannot find module '@babel/runtime/helpers/extends'`
5. Clone (if necessary) and link the local `stream-chat-react` repo with `yarn link /path/to/stream-chat-react` (replacing /path/to with the correct path to your local stream-chat-react repo).
6. In the local stream-chat-react repo, `yarn add @babel/runtime` (@babel/runtime should be a direct dependency not a dev dependency)
7. Running `yarn test` or `yarn build` in the `stream-chat-react-yarn-modern-bug-repro-repo` will now work.

**Expected behaviour**

I expect that the required @babel/runtime helpers are provided by the 3rd party library I'm using so that my test and build don't fail. I shouldn't have to provide this in my project if I'm not using @babel/plugin-transform-runtime` for any of my own transpiled code.

**Screenshots**

N/A

**Package version**
- stream-chat-react: 11.1.2
- stream-chat-css: n/a
- stream-chat-js: n/a

**Desktop (please complete the following information):**

- OS: [e.g. iOS]
- Browser [e.g. chrome, safari]
- Version [e.g. 22]

**Smartphone (please complete the following information):**

- Device: [e.g. iPhone6]
- OS: [e.g. iOS8.1]
- Browser [e.g. stock browser, safari]
- Version [e.g. 22]

**Additional context**

Add any other context about the problem here.

--- a/package.json
+++ b/package.json
@@ -58,6 +58,7 @@
     "getstream.io"
   ],
   "dependencies": {
+    "@babel/runtime": "^7.23.6",
     "@braintree/sanitize-url": "^6.0.4",
     "@popperjs/core": "^2.11.5",
     "clsx": "^2.0.0",

